### ISSUE TITLE
Loading latest release into Pharo 10 has stopped working

### ISSUE THREAD

**Issue by StewMacLean** [ObjectProfile/Roassal3#450]
As part of my image rebuild script I load Roassal into a fresh Pharo 10 image using:

[Metacello new
    		baseline: 'Roassal3';
    		repository: 'github://ObjectProfile/Roassal3:v0.9.9f';
    		load] on: MCMergeOrLoadWarning do: [: warning | warning load].

There are a few progress flickers and then all goes quiet - the UI is unresponsive...

If I wait it appears that most of the code has been installed, but the load process doesn't complete, so my build script grinds to a halt. There are a few walkbacks in the bowels of Iceberg....

Strange, as I'm still using a pristine Pharo 10 image, the same Roassal release, as when it was working a week or so ago.

Maybe Metacello does some updates before executing?

**Comment by akevalion** [comment]
I have test it and I can not reproduce it

roassal3 works on my machine

<img width="970" alt="image" src="https://user-images.githubusercontent.com/10532890/171192547-13be4f1b-ec1c-4743-bef2-f12a3afbb8c9.png">

Maybe this is related to the vm, I can suggest to you download last version of pharo launcher and delete your vm folder then, launcher is going to download it again

**Comment by akevalion** [comment]
@StewMacLean Let me know if reinstalling the launcher works

**Comment by akevalion** [comment]
Please try
```
mkdir roassalerror
cd roassalerror
curl -L https://get.pharo.org/64/ | bash
,/pharo
```
```st
[Metacello new
baseline: 'Roassal3';
repository: 'github://ObjectProfile/Roassal3:v0.9.9f';
load] on: MCMergeOrLoadWarning do: [: warning | warning load].
```

**Comment by StewMacLean** [comment]
Hi Milton, I've tried using the Launcher, but I have the same problem with
Official P10.

However, I have discovered the output below in the Transcript. (was in my
previous attempts as well, but just didn't see it!).

Looks like an authentication error - I did set my keys ages ago when I was
trying to use Iceberg on my private repo, but gave up cause it didn't work.
AFAIK, I haven't changed anything recently with regard to authentication.
The only thing I can think of is maybe the GitHub client application?

As this is a public repository I would not have expected any authentication
to occur?

Cheers,

Stewart

[Metacello new
baseline: 'Roassal3';
repository: 'github://ObjectProfile/Roassal3:v0.9.9f';
load] on: MCMergeOrLoadWarning do: [: warning | warning load].aborting


I got an error while cloning: There was an authentication error while
trying to execute the operation: .
This happens usually because you didn't provide a valid set of credentials.
You may fix this problem in different ways:

1. adding your keys to ssh-agent, executing ssh-add ~/.ssh/id_rsa in your
command line.
2. adding your keys in settings (open settings browser search for "Use
custom SSH keys" and
add your public and private keys).
3. using HTTPS instead SSH (Just use an url in the form HTTPS://etc.git). I
will try to clone the HTTPS variant.
aborting
I got an error while cloning: There was an authentication error while
trying to execute the operation: .
This happens usually because you didn't provide a valid set of credentials.

**Comment by akevalion** [comment]
I see in that case you will need to use token authentication
- In iceberg press button Settings
- In Settings browser check for button `Edit`
- In Current Iceberg credential window use `Token`
- You will need to access with your github account and create a token
- https://docs.github.com/es/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token
![image](https://user-images.githubusercontent.com/10532890/171293862-ade81bc3-4dc5-42ee-819c-8e517a21b1dd.png)

**Comment by StewMacLean** [comment]
The thing is, I'm downloading from the public Roassal repository. 

Does this mean that some thing has changed regarding authentication for Roassal?

I do not want to configure authentication unless I have to. What has changed to cause this problem with authentication in a fresh image? Is it a setting on my machine, or with the Roassal repository?

**Comment by akevalion** [comment]
I think this issue is related to iceberg,
Could you try with other repository.

```st
Metacello new
  baseline: 'DataFrame';
  repository: 'github://PolyMathOrg/DataFrame:v2.0/src';
  load.
```

**Comment by StewMacLean** [comment]
Yes, that is what I suspect too. I know it stores credentials in a Fuel file somewhere, maybe this is confusing it? But I'm not sure why it is Roassal specific, and what I've done to change the behaviour. It is an unfortunate design where authentication is so intertwined with the essential functionality of downloading data.

The DataFrame loads without a problem. As do other packages such as OSSubprocess. Thanks again for looking into this.

**Comment by StewMacLean** [comment]
Hi Milton,

I've finally got around to digging into this a bit deeper.

It appears to be failing on a particular package, namely Roassal3-Layouts-Util. For some reason an authentication error happens just for this, and then it then retry's by loading everything again until the same error occurs on the same package. It then loops indefinitely.

I've created a video that demonstrates this happening in a fresh image. Please see attached (google link, it is rather large).

Also attached is a screen shot of the packages that have been successfully downloaded prior to the error.  About 29 out of 39.

Please let me know if there's further diagnostics that you need. 

Thanks,

Stewart

https://drive.google.com/file/d/1gNeC8pBvIrIV4RqIhisiZZBjOBjBip8x/view

<img width="1062" alt="Partially downloaded" src="https://user-images.githubusercontent.com/97324099/174885490-559534e7-a2b7-442a-a4f2-4b7d9ce88dd9.png">

PS Apologies if you've got this twice - the email reply didn't appear here, so I've posted again.

**Comment by StewMacLean** [comment]
Hi Milton,

Well I've dived even deeper into this and found the problem!

There is a system wide registry and the setting shareRepositoriesBetweenImages had been set to true.

Within the fresh image, when loading packages the error handling was getting screwed up with duplicate packages. That's an Iceberg bug.

The solution was to set this to false (**IceLibgitRepository shareRepositoriesBetweenImages: false**), and everything works fine! Not sure where the Authentication error comes from, but my Image Builder is back up and running!

Thanks for you assistance - closing this off :)

Cheers,

Stewart

**Comment by akevalion** [comment]
@StewMacLean that is a good new!

**Comment by StewMacLean** [comment]
Hi Milton,

I've finally got around to digging into this a bit deeper.

It appears to be failing on a particular package, namely
*Roassal3-Layouts-Util*. For some reason an authentication error happens
just for this, it then retry's. It starts again at the beginning and loops
indefinitely.

I've created a video that demonstrates this happening in a fresh image.
Please see attached (google link, it is rather large).

Also attached is a screen shot of the packages that have been
successfully downloaded prior to the error.

Please let me know if there's further diagnostics that you need.

Thanks,

Stewart
 Infinite loop loading.mov
<https://drive.google.com/file/d/1gNeC8pBvIrIV4RqIhisiZZBjOBjBip8x/view?usp=drive_web>

**Comment by akevalion** [comment]
Hello StewMacLean, thanks for the video, it still not clear for me, why do you have this error. I am not able to reproduce it. Then when you have time we can have a video call videa meet or discord if you agree.

**Comment by StewMacLean** [comment]
Hi Milton,

After a lot of investigation I established the cause of the infinite loop
is within Iceberg authentication and the nature of my various git
repositories.

Once I executed before loading Roassal:

*IceLibgitRepository shareRepositoriesBetweenImages: false.*

This stopped the infinite loop. It was a while ago now, so I'm sorry I've
forgotten the details...

Hope this helps!

Cheers,

Stewart
